packages/web-app-draw-io/src/App.vue

```diff
@@ -1,261 +1,161 @@
 <template>
-  <main>
-    <div v-if="loading" class="oc-position-center">
-      <oc-spinner size="xlarge" />
-      <p v-translate class="oc-invisible">Loading media</p>
-    </div>
-    <iframe
-      v-else
-      id="drawio-editor"
-      ref="drawIoEditor"
-      :src="iframeSource"
-      :title="$gettext('Draw.io editor')"
-    />
-  </main>
+  <iframe
+    id="drawio-editor"
+    ref="drawIoEditor"
+    :src="iframeSource"
+    :title="$gettext('Draw.io editor')"
+  />
 </template>
 <script lang="ts">
-import { DateTime } from 'luxon'
-import { basename } from 'path'
 import qs from 'qs'
-import { defineComponent, unref } from 'vue'
-import { mapActions } from 'vuex'
-import { DavPermission, DavProperty } from 'web-client/src/webdav/constants'
-import { useAppDefaults, useStore } from 'web-pkg/src/composables'
+import {
+  defineComponent,
+  unref,
+  PropType,
+  computed,
+  onMounted,
+  onBeforeUnmount,
+  VNodeRef,
+  ref,
+  watch,
+  nextTick
+} from 'vue'
+import { Resource } from 'web-client/src'
+import { AppConfigObject } from 'web-pkg/src/apps'
 
 export default defineComponent({
   name: 'DrawIoEditor',
-  setup() {
-    const store = useStore()
-    const isAutoSaveEnabled = store.getters.configuration.options.editor.autosaveEnabled
-
-    return {
-      ...useAppDefaults({
-        applicationId: 'draw-io'
-      }),
-      isAutoSaveEnabled
+  props: {
+    resource: {
+      type: Object as PropType<Resource>,
+      required: true
+    },
+    applicationConfig: {
+      type: Object as PropType<AppConfigObject>,
+      required: true
+    },
+    currentContent: {
+      type: String,
+      required: true
+    },
+    isReadOnly: {
+      type: Boolean,
+      required: true
+    },
+    isDirty: {
+      type: Boolean,
+      required: true
     }
   },
-  data: () => ({
-    loading: true,
-    filePath: '',
-    fileExtension: '',
-    isReadOnly: null,
-    currentETag: null
-  }),
-  computed: {
-    config() {
-      const { url = 'https://embed.diagrams.net', theme = 'minimal' } = this.applicationConfig
+  emits: ['update:currentContent', 'save', 'close'],
+  setup(props, { emit }) {
+    const drawIoEditor: VNodeRef = ref()
+
+    const config = computed(() => {
```

Comment:
nitpick or taste and not sure whats correct, you could make a `const _config = unref(config)` and save some unrefs (but I ask this here because I'm not sure myself :) )

Reply:
Not really, that would break reactivity